apps/veil/src/pages/tournament/server/delegator-history.ts

```diff
@@ -32,162 +31,118 @@ export interface TournamentDelegatorHistoryResponse {
+    .orderBy(sortKey, sortDirection)
+    .orderBy('power', 'desc')
+    .limit(limit)
+    .offset(limit * (page - 1));
+
+  // calculate the total amount of all items in the filtered table with a sum of all rewards
+  const totalQuery = pindexerDb
+    .selectFrom(groupedByEpoch.as('rewards'))
+    .select(eb => [eb.fn.countAll().as('total_items'), eb.fn.sum('reward').as('total_reward')]);
+
+  const [delegatorHistory, total] = await Promise.all([
+    sorted.execute(),
+    totalQuery.executeTakeFirst(),
+  ]);
 
   return {
-    paginatedResults,
-    totalItems: results.length,
-    totalReward,
+    delegatorHistory,
+    totalRewards: Number(total?.total_reward) || 0,
+    totalItems: Number(total?.total_items) || 0,
   };
-}
+};
 
 export async function POST(
   req: NextRequest,
 ): Promise<NextResponse<Serialized<TournamentDelegatorHistoryResponse | { error: string }>>> {
-  const data = (await req.json()) as DelegatorHistoryRequest;
-  const { epochs, address, limit = BASE_LIMIT, page = BASE_PAGE, sortDirection = 'desc' } = data;
+  const params = await getBodyParams(req);
+  if (typeof params === 'string') {
+    return NextResponse.json({ error: params }, { status: 400 });
+  }
 
-  const targetAddress = bech32mAddress(address);
+  const { delegatorHistory, totalRewards, totalItems } =
+    await tournamentDelegatorHistoryQuery(params);
 
-  // Execute the queries for all epochs concurrently
-  const queryPromises = [...epochs].map(async epoch => {
-    const data = await tournamentDelegatorHistoryQuery({
-      epoch: Number(epoch),
-    });
+  const history = delegatorHistory.map<LqtDelegatorHistoryData>(item => {
     return {
-      epoch,
-      data,
+      epoch: item.epoch,
+      power: Number(item.power),
+      reward: Number(item.reward),
+      address: new Address({ inner: item.address }),
+      asset_id: new AssetId({ inner: Uint8Array.from(item.asset_id) }),
     };
   });
 
-  const epochResults = await Promise.all(queryPromises);
-
-  // Return the delegator’s epoch history and the total reward accumulated across all epochs.
-  const { paginatedResults, totalItems, totalReward } = processEpochResults(
-    epochResults,
-    targetAddress,
-    address,
-    limit,
-    page,
-    sortDirection,
-  );
-
-  const chainId = process.env['PENUMBRA_CHAIN_ID'];
-  if (!chainId) {
-    return NextResponse.json({ error: 'Error: PENUMBRA_CHAIN_ID is not set' }, { status: 500 });
-  }
-
-  const registryClient = new ChainRegistryClient();
-  const registry = await registryClient.remote.get(chainId);
-
-  // Map only the paginated results with metadata
-  const mapped = await Promise.all(
-    paginatedResults.map(item => {
-      // TODO: remove hardcoded staking asset metadata when registry is fixed.
-      const stakingAssetId = registryClient.bundled.globals().stakingAssetId;
-      const asset_id = new AssetId({ inner: Uint8Array.from(item.asset_id) });
-      const metadata = registry.tryGetMetadata(stakingAssetId);
```

Comment:
removed metadata searching from the backend as frontend duplicates this work anyway